### PR TITLE
Apply Calibtool for FCN

### PR DESCRIPTION
**Pull request by wuxun-zhang**
Pending on [#893](https://github.com/dmlc/gluon-cv/pull/893)

This PR is to add calibtool for FCN model firstly. More segmentation models are under working.

@pengzhao-intel @xinyu-intel @zhreshold

### CODE REVIEW

**Comment by mli** [comment]
Job PR-903-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-903/1/index.html 
Code coverage of this PR:  vs. Master:

**Comment by xinyu-intel** [comment]
@wuxun-zhang PR #893 merged, please rebase and add a test for this model in `test_calib_models()`

File: scripts/segmentation/test.py [hunk 1]
```diff
@@ -18,10 +20,14 @@
 def parse_args():
     parser = argparse.ArgumentParser(description='Validation on Segmentation model')
     # model and dataset
-    parser.add_argument('--model', type=str, default='fcn',
-                        help='model name (default: fcn)')
+    parser.add_argument('--network', type=str, default='fcn',
```
Comment (xinyu-intel) [inline scripts/segmentation/test.py]: I think we should not break the user interface.

File: scripts/segmentation/test.py [hunk 2]
```diff
@@ -86,8 +113,8 @@ def test(model, args, input_transform):
     test_data = gluon.data.DataLoader(
         testset, args.batch_size, shuffle=False, last_batch='keep',
         batchify_fn=ms_batchify_fn, num_workers=args.workers)
-    print(model)
-    evaluator = MultiEvalModel(model, testset.num_class, ctx_list=args.ctx)
+    print(net)
```
Comment (xinyu-intel) [inline scripts/segmentation/test.py]: can we keep the original variable names and introduce only we need?

File: scripts/segmentation/test.py [hunk 3]
```diff
@@ -188,56 +194,110 @@ def benchmarking(model, args):
 
 if __name__ == "__main__":
     args = parse_args()
+    logging.basicConfig()
+    logger = logging.getLogger('logger')
+    logger.setLevel(logging.INFO)
+    logging.info(args)
 
     withQuantization = False
-    model_prefix = args.model + '_' + args.backbone
```
Comment (xinyu-intel) [inline scripts/segmentation/test.py]: same

File: scripts/segmentation/test.py [hunk 4]
```diff
@@ -207,37 +217,87 @@ def benchmarking(model, args):
...
     # image transform
     input_transform = transforms.Compose([
         transforms.ToTensor(),
         transforms.Normalize([.485, .456, .406], [.229, .224, .225]),
     ])
 
-    if not args.benchmark:
-        if '_int8' in model_prefix:
-            test_quantization(model, args, input_transform)
+    if args.calibration or '_int8' in model_prefix:
```
Comment (wuxun-zhang) [inline scripts/segmentation/test.py]: @xinyu-intel I'm not sure if it is appropriate to put `calibration` part into `test_quantization` function. Then, changes for common part will become much smaller (only have changes in `test_quantization` function).

**Comment by mli** [comment]
Job PR-903-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-903/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-903-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-903/3/index.html 
Code coverage of this PR:  vs. Master:

**Comment by pengzhao-intel** [comment]
The new chart with FCN looks good!

File: scripts/segmentation/test.py [hunk 5]
```diff
@@ -207,37 +217,87 @@ def benchmarking(model, args):
...
+                                 exclude_layers=exclude_sym_layer, num_calib_examples=args.batch_size * args.num_calib_batches,
+                                 exclude_layers_match=exclude_match_layer, ctx=args.ctx[0], logger=logger)
+            dir_path = os.path.dirname(os.path.realpath(__file__))
+            dst_dir = os.path.join(dir_path, 'model')
+            if not os.path.isdir(dst_dir):
+                os.mkdir(dst_dir)
+            prefix = os.path.join(dst_dir, model_prefix + '-quantized-' + args.calib_mode)
+            logger.info('Saving quantized model at %s' % dst_dir)
+            model.export(prefix, epoch=0)
+            sys.exit()
```
Comment (zhreshold) [inline scripts/segmentation/test.py]: why sys.exit?

**Comment by zhreshold** [comment]
The table rendered is a bit too wide on my laptop: http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-903/3/build/examples_deployment/int8_inference.html#performance

File: scripts/segmentation/test.py (hunk 5 above)
Comment (wuxun-zhang) [reply scripts/segmentation/test.py]: We assume that users run this script either in `calibration` or `test` mode. That is, users can get quantized models from their own FP32 models in `calibration` mode firstly and then run inference with these saved models in `test` mode.

**Comment by mli** [comment]
Job PR-903-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-903/4/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-903-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-903/5/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**